src/libostree/ostree-repo.c

```diff
@@ -210,62 +221,51 @@ free_repo_lock_table (gpointer data)
     }
 }
 
+/* When pushing lock states, it's clearly known what the current and next lock
+ * state are. However, when popping lock states we need to what the lock state
+ * that was pushed by the current thread was so we know what state to drop. A
+ * stack is kept per thread for this.
+ *
+ * In order to allow a single thread to have multiple OstreeRepos, a hash
+ * table with the OstreeRepo as key and GPtrArray as value is maintained in
+ * thread local storage via GPrivate. The array represents that thread's lock
+ * stack for a particular OstreeRepo.
+ */
 static GPrivate repo_lock_table = G_PRIVATE_INIT (free_repo_lock_table);
 
-typedef struct {
-  int fd;
-  GQueue stack;
-} OstreeRepoLock;
-
 typedef struct {
   guint len;
   int state;
   const char *name;
 } OstreeRepoLockInfo;
 
 static void
-repo_lock_info (OstreeRepoLock *lock, OstreeRepoLockInfo *out_info)
+repo_lock_info (OstreeRepo *self, OstreeRepoLockInfo *out_info)
```

Comment:
This seems to internally assume that `self->lock` details are unchanged throughout body execution, i.e. that the function can be called if and only if run the caller already holds the related mutex.
If that is indeed the intended design, it may be nice to let the caller be explicitly aware of that, e.g. by requiring a mutex locker/guard as an input argument.

Reply:
It's just a helper function to not have a bunch of adhoc conversions of lock state to reporting info, but I guess since `GMutexLocker` is already in use it could be used to ensure the mutex is taken.